llama-index-integrations/llms/llama-index-llms-qianfan/llama_index/llms/qianfan/base.py

```diff
@@ -0,0 +1,376 @@
+            and the previous messages are the historical chat information. The number of
+            members must be odd, and the role value of the odd-numbered messages must be
+            "user", while the role value of the even-numbered messages must be "assistant".
+        :return: The ChatResponse object.
+        """
+        request = build_chat_request(stream=False, messages=messages, **kwargs)
+        resp_dict = await self._client.apost(self.endpoint_url, json=request.dict())
+        return parse_chat_response(resp_dict)
+
+    @llm_chat_callback()
+    def stream_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponseGen:
+        """
+        Request a chat, and the response is returned in a stream.
+
+        :param messages: The chat message list. The last message is the current request,
+            and the previous messages are the historical chat information. The number of
+            members must be odd, and the role value of the odd-numbered messages must be
+            "user", while the role value of the even-numbered messages must be "assistant".
+        :return: A ChatResponseGen object, which is a generator of ChatResponse.
+        """
+        request = build_chat_request(stream=True, messages=messages, **kwargs)
+
+        def gen():
+            resp_dict_iter = self._client.post_reply_stream(
+                self.endpoint_url, json=request.dict()
+            )
+            yield from parse_stream_chat_response(resp_dict_iter)
+
+        return gen()
+
+    @llm_chat_callback()
+    async def astream_chat(
+        self,
+        messages: Sequence[ChatMessage],
+        **kwargs: Any,
+    ) -> ChatResponseAsyncGen:
+        """
+        Asynchronous request a chat, and the response is returned in a stream.
+
+        :param messages: The chat message list. The last message is the current request,
+            and the previous messages are the historical chat information. The number of
+            members must be odd, and the role value of the odd-numbered messages must be
+            "user", while the role value of the even-numbered messages must be "assistant".
+        :return: A ChatResponseAsyncGen object, which is a asynchronous generator of ChatResponse.
+        """
+        request = build_chat_request(stream=True, messages=messages, **kwargs)
+
+        async def gen():
+            resp_dict_iter = self._client.apost_reply_stream(
+                self.endpoint_url, json=request.dict()
+            )
+            async for part in aparse_stream_chat_response(resp_dict_iter):
+                yield part
+
+        return gen()
+
+    @llm_chat_callback()
+    def complete(
+        self, prompt: str, formatted: bool = False, **kwargs: Any
+    ) -> CompletionResponse:
+        """
+        Not implemented.
+        """
+        raise NotImplementedError("complete is not supported for Qianfan-LLM")
```

Comment:
we actually have a wrapper to wrap a complete request into a chat request, you can see the OpenAI class for an example. Up to you though!